thor/tests/e2e.rs

```diff
@@ -410,3 +440,213 @@ async fn e2e_force_close_after_updates() {
+    let (alice_wallet, bob_wallet) = make_wallets(&bitcoind, fund_amount_alice, fund_amount_bob)
+        .await
+        .unwrap();
+    let (mut alice_transport, mut bob_transport) = make_transports();
+
+    let alice_create = Channel::create(
+        &mut alice_transport,
+        &alice_wallet,
+        balance_alice,
+        time_lock,
+    );
+    let bob_create = Channel::create(&mut bob_transport, &bob_wallet, balance_bob, time_lock);
+
+    let (mut alice_channel, mut bob_channel) = futures::future::try_join(alice_create, bob_create)
+        .await
+        .unwrap();
+
+    let after_open_balance_alice = alice_wallet.0.balance().await.unwrap();
+    let after_open_balance_bob = bob_wallet.0.balance().await.unwrap();
+
+    let (secret, point) = random_key_pair();
+    let ptlc_amount = Amount::from_btc(0.5).unwrap();
+
+    let (alpha_absolute_expiry, TX_s_time_lock, ptlc_redeem_time_lock) = (1_598_875_222, 1, 1);
+
+    let alice_add_ptlc = alice_channel.add_ptlc_alice(
+        &mut alice_transport,
+        &alice_wallet,
+        ptlc_amount,
+        secret,
+        alpha_absolute_expiry,
+        TX_s_time_lock,
+        ptlc_redeem_time_lock,
+    );
+
+    let mut bob_add_ptlc = bob_channel.add_ptlc_bob(
+        &mut bob_transport,
+        ptlc_amount,
+        point,
+        alpha_absolute_expiry,
+        TX_s_time_lock,
+        ptlc_redeem_time_lock,
+    );
+    let bob_swap = async {
+        match bob_add_ptlc.async_resume().await {
+            GeneratorState::Yielded(_secret) => {
+                // TODO: Redeem alpha asset
+
+                // Do not perform final update
+                Ok(())
+            }
+            GeneratorState::Complete(Err(e)) => panic!("{}", e),
+            GeneratorState::Complete(Ok(())) => panic!("did not yield secret"),
+        }
+    };
+
+    futures::future::try_join(alice_add_ptlc, bob_swap)
+        .await
+        .unwrap();
+
+    let after_close_balance_alice = alice_wallet.0.balance().await.unwrap();
+    let after_close_balance_bob = bob_wallet.0.balance().await.unwrap();
+
+    // We pay half a `thor::TX_FEE` per output in fees for each transaction after
+    // the `FundingTransaction`. Force closing the channel requires
+    // publishing two transactions: a `CommitTransaction` and a `SplitTransaction`,
+    // so each party pays a full `thor::TX_FEE`, which is deducted from their
+    // output.
+    let fee_deduction_per_balance_output = Amount::from_sat(thor::TX_FEE);
+
+    // Additionally, Alice pays an extra `thor::TX_FEE` to be able to redeem the
+    // PTLC output.
+    let fee_deduction_for_ptlc_redeem = Amount::from_sat(thor::TX_FEE);
+
+    assert_eq!(
+        after_close_balance_alice,
+        after_open_balance_alice + fund_amount_alice + ptlc_amount
+            - fee_deduction_per_balance_output - fee_deduction_for_ptlc_redeem,
+        "Balance after closing channel should equal balance after opening plus PTLC amount, minus transaction fees"
```

Comment:
```suggestion
        "Balance after closing channel should equal balance after opening plus PTLC amount, plus fund amount, minus transaction fees"
```
I found this assertion unintuitive. I think it is safer (in the regards of not loosing money) to that the diff of  balance **before** opening the channel and **after** closing the channel matches the payments done over the channel.

For example. if twice the `fund_amount` is taken from the balance to open the channel, then this assert would not catch the bug and ultimately the user would lose money.

Reply:
I agree. I thought it may be difficult to achieve what you describe because we don't control the funding transaction fee, but maybe I was wrong. We should fix it in a bunch of tests, so maybe we could leave it as a TODO and fix it in a follow-up PR.

Reply:
Have a look at my PR: #49, I did that with @bonomat's help.

Reply:
Not gonna do it now, but we should definitely align all tests to do the same thing at some point.